src/Sylius/Component/Core/Provider/ProductVariantsPricesProvider.php

```diff
@@ -53,7 +55,15 @@ private function constructOptionsMap(ProductVariantInterface $variant, ChannelIn
             $optionMap[$option->getOptionCode()] = $option->getCode();
         }
 
-        $optionMap['value'] = $this->productVariantPriceCalculator->calculate($variant, ['channel' => $channel]);
+        $price = $this->productVariantPriceCalculator->calculate($variant, ['channel' => $channel]);
+        $optionMap['value'] = $price;
+
+        Assert::isInstanceOf($this->productVariantPriceCalculator, ProductVariantPricesCalculatorInterface::class);
```

Comment:
Maybe it would be safer to use `if` rather than assertion here? **Technically** it's still possible to pass the old `ProductVariantPriceCalculatorInterface` in the constructor (for example if someone overrode the service declaration)